data/app.d/historical_data_parser.py

```diff
@@ -10,11 +10,45 @@ class MissingNullsJsonWriter:
     Attributes:
         rows (list<dict>): A list of dictionaries representing each row to write
         columns (dict): A dictionary representing the columns to write
+        datetime_converter (method): A method that takes a string and converts it
+            to a Deephaven DateTime object. If the string isn't convertable to a
+            DateTime object, None should be returned
     """
 
-    def __init__(self):
+    def __init__(self, datetime_converter=None):
         self.rows = []
         self.columns = {}
+        self.datetime_converter = datetime_converter
+
+    def _build_columns(self):
+        """
+        Builds the columns attribute based on the current rows
+
+        Returns:
+            None
+        """
+        self.columns = {}
+
+        for row in self.rows:
+            for column in row:
+                if not (column in self.columns):
```

Comment:
this is going to guess the column type based on the first row for each column? sounds fine for sportradar data, just confirming

Reply:
Yes that's correct